crates/oxc_linter/src/rules/jest/no_deprecated_functions.rs

```diff
@@ -0,0 +1,180 @@
+    ///
+    /// #### `jest.addMatchers`
+    /// This function was replaced with `expect.extend` in Jest 17 and removed in Jest 27.
+    ///
+    /// #### `require.requireActual` & `require.requireMock`
+    /// These functions were replaced in Jest 21 and removed in Jest 26.
+    ///
+    /// Originally, the `requireActual` & `requireMock` the `requireActual`&
+    /// `requireMock` functions were placed onto the `require` function.
+    ///
+    /// These functions were later moved onto the `jest` object in order to be easier
+    /// for type checkers to handle, and their use via `require` deprecated. Finally,
+    /// the release of Jest 26 saw them removed from the `require` function altogether.
+    ///
+    /// #### `jest.runTimersToTime`
+    /// This function was renamed to `advanceTimersByTime` in Jest 22 and removed in Jest 27.
+    ///
+    /// #### `jest.genMockFromModule`
+    /// This function was renamed to `createMockFromModule` in Jest 26, and is scheduled for removal in Jest 30.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// While typically these deprecated functions are kept in the codebase for a number
+    /// of majors, eventually they are removed completely.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// jest.resetModuleRegistry // since Jest 15
+    /// jest.addMatchers // since Jest 17
+    /// ```
+    NoDeprecatedFunctions,
+    style,
+);
+
+const DEPRECATED_FUNCTIONS_MAP: Map<&'static str, (usize, &'static str)> = phf_map! {
+    "jest.resetModuleRegistry" => (15, "jest.resetModules"),
+    "jest.addMatchers" => (17, "expect.extend"),
+    "require.requireMock" => (21, "jest.requireMock"),
+    "require.requireActual" => (21, "jest.requireMock"),
+    "jest.runTimersToTime" => (22, "jest.advanceTimersByTime"),
+    "jest.genMockFromModule" => (26, "jest.createMockFromModule"),
+};
+
+impl Rule for NoDeprecatedFunctions {
+    fn from_configuration(value: serde_json::Value) -> Self {
+        let version = value
+            .get(0)
+            .and_then(|v| v.get("jest"))
+            .and_then(|v| v.get("version"))
+            .and_then(|v| serde_json::Value::as_str(v))
+            // Todo: Fixed Me
+            // Currently set the default version to the (maybe) latest, to help to find more problems in
+            // the codebase. In the future, the version should come from the cli option or the config files,
+            // such as `package.json` or `eslint.config.js`.
+            .unwrap_or("29");
+
+        let major: Vec<&str> = version.split('.').collect();
+
+        Self { jest: JestConfig { version: major[0].to_string() } }
+    }
+
+    fn run<'a>(&self, node: &oxc_semantic::AstNode<'a>, ctx: &LintContext<'a>) {
+        let AstKind::MemberExpression(mem_expr) = node.kind() else {
+            return;
+        };
+        let mut chain: Vec<Cow<'a, str>> = Vec::new();
+        chain.extend(get_node_name_vec(mem_expr.object()));
```

Comment:
Is this will be slow to get all `MeberExpression`'s node names? Do you want to try a better way?

Reply:
Done.